store/mockstore/mocktikv/cop_handler_dag.go

```diff
@@ -471,21 +474,37 @@ func (mock *mockCopStreamClient) Recv() (*coprocessor.Response, error) {
 	return &resp, nil
 }
 
-func (mock *mockCopStreamClient) readBlockFromExecutor() (tipb.Chunk, bool, error) {
+func (mock *mockCopStreamClient) readBlockFromExecutor(counts []int64) (tipb.Chunk, bool, *coprocessor.KeyRange, error) {
 	var chunk tipb.Chunk
+	var ran coprocessor.KeyRange
+	var finish bool
+	mock.exec.ResetCount()
+	ran.Start = mock.exec.Cursor()
 	for count := 0; count < rowsPerChunk; count++ {
 		row, err := mock.exec.Next(mock.ctx)
 		if err != nil {
-			return chunk, false, errors.Trace(err)
+			return chunk, false, &ran, errors.Trace(err)
 		}
 		if row == nil {
-			return chunk, true, nil
+			finish = true
+			break
 		}
 		for _, offset := range mock.req.OutputOffsets {
 			chunk.RowsData = append(chunk.RowsData, row[offset]...)
 		}
 	}
-	return chunk, false, nil
+
+	ran.End = mock.exec.Cursor()
```

Comment:
For desc scan, the start may be larger than end? But from the [comment](https://github.com/pingcap/kvproto/blob/master/proto/coprocessor.proto#L14) start should always smaller than end?